src/main/kotlin/io/emeraldpay/dshackle/config/ChainsConfig.kt

```diff
@@ -0,0 +1,63 @@
+package io.emeraldpay.dshackle.config
+
+import io.emeraldpay.dshackle.Chain
+import java.lang.IllegalStateException
+
+data class ChainsConfig(private val chains: Map<Chain, RawChainConfig>, val currentDefault: RawChainConfig?) {
+    companion object {
+        @JvmStatic
+        fun default(): ChainsConfig = ChainsConfig(emptyMap(), RawChainConfig.default())
+    }
+
+    data class RawChainConfig(val syncingLagSize: Int?, val laggingLagSize: Int?) {
+        companion object {
+            @JvmStatic
+            fun default() = RawChainConfig(6, 1)
+        }
+    }
+
+    data class ChainConfig(val syncingLagSize: Int, val laggingLagSize: Int) {
```

Comment:
Are there any differences with `RawChainConfig`? They look identically except notnull types,  maybe leave only one of them?

Reply:
I want to have nulls in merging and then create a non-nullable version for end-user code.